webapp/packages/plugin-data-import/src/DataImportBootstrap.ts

```diff
@@ -0,0 +1,145 @@
+  constructor(
+    private readonly menuService: MenuService,
+    private readonly actionService: ActionService,
+    private readonly notificationService: NotificationService,
+    private readonly commonDialogService: CommonDialogService,
+    private readonly dataImportService: DataImportService,
+  ) {
+    super();
+  }
+
+  register() {
+    this.actionService.addHandler({
+      id: 'data-import-base-handler',
+      isActionApplicable(context, action) {
+        const menu = context.hasValue(DATA_CONTEXT_MENU, DATA_VIEWER_DATA_MODEL_ACTIONS_MENU);
+        const model = context.tryGet(DATA_CONTEXT_DV_DDM);
+        const resultIndex = context.tryGet(DATA_CONTEXT_DV_DDM_RESULT_INDEX);
+
+        if (!menu || !model || resultIndex === undefined) {
+          return false;
+        }
+
+        if (action === ACTION_IMPORT) {
+          if (model.isReadonly(resultIndex)) {
+            return false;
+          }
+
+          const editor = model.source.getActionImplementation(resultIndex, DatabaseEditAction);
+          return editor?.hasFeature('add') === true;
+        }
+
+        return [ACTION_IMPORT].includes(action);
+      },
+      isDisabled(context) {
+        const model = context.get(DATA_CONTEXT_DV_DDM);
+        const resultIndex = context.get(DATA_CONTEXT_DV_DDM_RESULT_INDEX);
+
+        return model.isLoading() || model.isDisabled(resultIndex) || !model.getResult(resultIndex);
+      },
+      getActionInfo(_, action) {
+        if (action === ACTION_IMPORT) {
+          return { ...action.info, icon: '/icons/data-import.png' };
+        }
+
+        return action.info;
+      },
+      handler: async (context, action) => {
+        const model = context.get(DATA_CONTEXT_DV_DDM);
+        const resultIndex = context.get(DATA_CONTEXT_DV_DDM_RESULT_INDEX);
+
+        if (action === ACTION_IMPORT) {
+          const result = model.getResult(resultIndex);
+
+          if (!result?.id) {
+            throw new Error('Result must be provided');
+          }
+
+          const executionContext = model.source.executionContext?.context;
+
+          if (!executionContext) {
+            throw new Error('Execution context must be provided');
+          }
+
+          await this.openImportDataDialog(
+            executionContext.connectionId,
+            executionContext.id,
+            executionContext.projectId,
+            result.id,
+            model.name ?? model.id,
+            undefined,
+            () => model.refresh(),
+          );
+        }
+      },
+    });
+
+    this.menuService.addCreator({
+      menus: [DATA_VIEWER_DATA_MODEL_ACTIONS_MENU],
+      isApplicable: () => !this.dataImportService.disabled,
+      getItems(_, items) {
+        return [...items, ACTION_IMPORT];
+      },
+      orderItems(_, items) {
+        const extracted = menuExtractItems(items, [ACTION_IMPORT]);
+        return [...items, ...extracted];
+      },
+    });
+  }
+
+  private async openImportDataDialog(
```

Comment:
this one also should be moved to the service so this bootstrap component not overloaded with the logic and we can also to reuse this function somehere we would need it